components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/ManagementApiParser.java

```diff
@@ -0,0 +1,154 @@
+        }
+        throw new ManagementApiUndefinedException("Management API not defined in " + getConfigurationFilePath());
+    }
+
+    public HashMap<String, char[]> getUserList() throws CarbonException, XMLStreamException, IOException,
+            UserStoreUndefinedException, ManagementApiUndefinedException {
+        if (Objects.nonNull(usersList)) {
+            return usersList;
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Parsing the UsetStore element from " + getConfigurationFilePath());
+        }
+        OMElement userStoreOM = getManagementApiElement().getFirstChildWithName(new QName("UserStore"));
+        if (Objects.nonNull(userStoreOM)) {
+            return populateUserList(userStoreOM.getFirstChildWithName(new QName("users")));
+        } else {
+            throw new UserStoreUndefinedException("UserStore tag not defined inside the Management API");
+        }
+    }
+
+    private static Iterator<OMElement> getInternalApisElement() throws IOException, CarbonException, XMLStreamException {
+        File mgtApiUserConfig = getConfigurationFile();
+        try (InputStream fileInputStream = new FileInputStream(mgtApiUserConfig)) {
+            OMElement documentElement = getOMElementFromFile(fileInputStream);
+            createSecretResolver(documentElement);
+
+            return documentElement.getFirstChildWithName(new QName("apis")).getChildrenWithName(new QName("api"));
+        }
+    }
+
+    /**
+     * Returns the document OMElement from the internal-apis.xml file.
+     *
+     * @param fileInputStream input stream of internal-apis.xml
+     * @return OMelement of internal-apis.xml
+     */
+    private static OMElement getOMElementFromFile(InputStream fileInputStream) throws CarbonException, XMLStreamException {
+        InputStream inputStream = MicroIntegratorBaseUtils.replaceSystemVariablesInXml(fileInputStream);
+        StAXOMBuilder builder = new StAXOMBuilder(inputStream);
+        return builder.getDocumentElement();
+    }
+
+    /**
+     * Populates the userList hashMap by user store OM element
+     * @return HashMap<String, char []> Map of credential pairs
+     */
+    private HashMap<String, char []> populateUserList(OMElement users) {
+        HashMap<String, char []> userList = new HashMap<>();
+        if (users != null) {
+            Iterator usersIterator = users.getChildrenWithName(new QName("user"));
+            if (usersIterator != null) {
+                while (usersIterator.hasNext()) {
+                    OMElement userElement = (OMElement) usersIterator.next();
+                    OMElement userNameElement = userElement.getFirstChildWithName(new QName("username"));
+                    OMElement passwordElement = userElement.getFirstChildWithName(new QName("password"));
+                    if (userNameElement != null && passwordElement != null) {
+                        userList.put(userNameElement.getText(), passwordElement.getText().toCharArray());
```

Comment:
Do we need to validate whether same userName is defined twice?

Reply:
done